logger.go

```diff
@@ -162,8 +160,8 @@ func (l *Logger) log(lvl Level, format *string, args ...interface{}) {
 	// methods, Info(), Fatal(), etc.
 	// ExtraCallDepth allows this to be extended further up the stack in case we
 	// are wrapping these methods, eg. to expose them package level
-	if l.haveBackend {
-		l.backend.Log(lvl, 2+l.ExtraCalldepth, record)
+	if l.backend.Load() != nil {
+		l.backend.Load().(LeveledBackend).Log(lvl, 2+l.ExtraCalldepth, record)
 		return
 	}
 
```

Comment:
This I would treat as separate to the "allow setting levels from other GoRoutines". I want to be able to set levels for modules from a single goroutine for everything. But I'm unsure if there's a need to be able to change the backend for a logger used on multiple goroutines. Personally, I prefer multiple loggers, one per routine, and never access them across routines.

Problem is this would only make `SetBackend` race-free. The exported properties `Method` and `ExtraCalldepth` wouldn't be, so not sure if that would cause some confusion?